packages/components/data-table-manager/src/data-table-manager.spec.js

```diff
@@ -0,0 +1,93 @@
+import { screen, render, fireEvent } from '../../../../test/test-utils';
+import DataTableManager from './data-table-manager';
+
+/* eslint-disable react/prop-types */
+const TestTable = (props) => (
+  <div>
+    <ul>
+      {props.columns.map((column) => (
+        <li key={column.key}>{column.label}</li>
+      ))}
+    </ul>
+  </div>
+);
+/* eslint-enable react/prop-types */
+
+const TestComponent = (props) => (
+  <DataTableManager {...props}>
+    <TestTable />
+  </DataTableManager>
+);
+
+const createTestProps = (custom = {}) => ({
+  columns: [
+    { key: 'title', label: 'Title' },
+    { key: 'country', label: 'Country' },
+  ],
+  displaySettings: {
+    disableDisplaySettings: false,
+  },
+  columnManager: {
+    disableColumnManager: false,
+  },
+  onSettingsChange: jest.fn(),
+  ...custom,
+});
+
+describe('rendering', () => {
+  it('should not render the dropdown if no settings options are passed', async () => {
+    const props = createTestProps({
+      displaySettings: undefined,
+      columnManager: undefined,
+    });
+    render(<TestComponent {...props} />);
+
+    expect(screen.queryByText('Table layout settings')).not.toBeInTheDocument();
+    expect(screen.queryByText('Column Manager')).not.toBeInTheDocument();
+    expect(
+      screen.queryByLabelText('Open table manager dropdown')
+    ).not.toBeInTheDocument();
+  });
+  it('should render the display settings panel when clicking on the dropdown option', async () => {
+    const props = createTestProps();
+    render(<TestComponent {...props} />);
+
+    expect(screen.queryByText('Table layout settings')).not.toBeInTheDocument();
+    expect(screen.queryByText('Column Manager')).not.toBeInTheDocument();
+
+    const selectDropdown = await screen.findByLabelText(
+      'Open table manager dropdown'
+    );
+
+    fireEvent.focus(selectDropdown);
+    fireEvent.change(selectDropdown, { target: { value: 'display' } });
+    fireEvent.keyDown(selectDropdown, { key: 'Enter', keyCode: 13, which: 13 });
+
+    await screen.findByText('Table layout settings');
+
+    expect(
+      screen.queryByLabelText('Select radio option: display full text')
+    ).toBeChecked();
+    expect(
+      screen.queryByLabelText('Select radio option: density default')
+    ).toBeChecked();
+  });
+  it('should render the column settings panel when clicking on the dropdown option', async () => {
+    const props = createTestProps();
+    render(<TestComponent {...props} />);
+
+    expect(screen.queryByText('Table layout settings')).not.toBeInTheDocument();
+    expect(screen.queryByText('Column Manager')).not.toBeInTheDocument();
+
+    const selectDropdown = await screen.findByLabelText(
+      'Open table manager dropdown'
+    );
+
+    fireEvent.focus(selectDropdown);
+    fireEvent.change(selectDropdown, { target: { value: 'column' } });
+    fireEvent.keyDown(selectDropdown, { key: 'Enter', keyCode: 13, which: 13 });
+
+    await screen.findByText('Column Manager');
+  });
```

Comment:
Alright, this is how I would like to write the tests.

I just added a bunch of tests to check for the initial rendering. Interaction tests will follow (possibly also in a separate PR to keep things smaller).